swift-sdk/Internal/AuthManager.swift

```diff
@@ -13,14 +13,20 @@ import Foundation
 
 class AuthManager: IterableInternalAuthManagerProtocol {
     init(onAuthTokenRequestedCallback: (() -> String?)?,
+         autoPushRegistration: Bool,
          localStorage: LocalStorageProtocol,
          dateProvider: DateProviderProtocol,
+         notificationStateProvider: NotificationStateProviderProtocol,
+         inAppManager: IterableInternalInAppManagerProtocol,
```

Comment:
The design looks strange. AuthManager should not have a dependency on inAppManager.

Reply:
The issue arises from the need to reset in-apps and registration. I'll figure out a new design for it.